### PR TITLE
(fix) Sound preferences: don't set m_settingsModified in update slots

### PR DESCRIPTION
**Pull request by ronso0**
Currently the `m_settingsModified` is set in `DlgPrefSound::slotUpdate()` which will cause a sound pause when pressing **Apply** in any pref page, because `DlgPrefSound::slotApply()` would reapply the current (unchanged) sound config.
https://github.com/mixxxdj/mixxx/pull/13070#discussion_r1667683929

Fixed by moving the keylock dual-threading checkbox update to a dedicated method and call then when necessary.

@acolombier please check if I got everything right.

### CODE REVIEW

**Review by acolombier — APPROVED** [review]
I haven't had a chance to test it due to time, but reviewed the code, and I can see no missing logic. Thanks for the fix!

**Review by daschuer — APPROVED** [review]
LGTM and thank you for the control proxy refactoring

File: src/preferences/dialog/dlgprefsound.cpp [hunk 1]
```diff
@@ -742,6 +751,9 @@ void DlgPrefSound::settingChanged() {
     m_settingsModified = true;
 
 #ifdef __RUBBERBAND__
+}
+
+void DlgPrefSound::updateKeylockDualThreadingCheckbox() {
```
Comment (fwcd) [inline src/preferences/dialog/dlgprefsound.cpp:756]: This `#ifdef` looks suspicious, does it work as intended? I'm getting some errors building for iOS (without Rubberband)...

Comment (fwcd) [reply src/preferences/dialog/dlgprefsound.cpp:756]: Seems to work, but the brace placement makes this confusing. I've fixed this and some related issues in

- https://github.com/mixxxdj/mixxx/pull/13577

**Pull request merged**